### PR TITLE
Improved RIS export

### PR DESCRIPTION
**Pull request by damien-git**
This PR adds additional tags to the RIS export. A number of them are based on new SolrMarc functions, so they are specific to this driver.

New tags:
- ST (short title)
- L4 (page URL as an attachment)
- AB (abstract)
- AV (notes)
- CN (call number)
- J2 (alternate title)

Updated tags:
- KW (keywords - from the topics/subjects; getTopics was not implemented before)
- DO (DOI identifier, removing markup from EDS)

There might be ways to improve the way this implementation works with other drivers...

TODO
- [x] When merging, note EDS record driver changes in changelog (replacement of getItemsSubjects with getAllSubjectHeadingsFlattened; fix to eliminate HTML from getCleanDOI)

### CODE REVIEW

File: module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: I believe that subfield c was intentionally omitted here to match the way that VuFind generally excludes statements of responsibility from title displays (since this information is instead displayed elsewhere and is not actually part of the title proper). Does it make sense to add it for the purposes of RIS? Do we really want non-title details in the J2 title field?

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: Typo: should be "concatenates".

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: What if you replaced the $indNum and $indValue parameters with a single array of indicator filters, and then any rows that match a filter will be allowed through? This way, instead of doing array merging in some of the functions below, you could request everything at once -- e.g.

```
        return array_merge(
            $this->getMarcFieldWithInd('520', null, '1', ''),
            $this->getMarcFieldWithInd('520', null, '1', '0'),
            $this->getMarcFieldWithInd('520', null, '1', '2'),
            $this->getMarcFieldWithInd('520', null, '1', '3'),
            $this->getMarcFieldWithInd('520', null, '1', '8'),
        );
```

becomes:

```
        return $this->getMarcFieldWithInd('520', null, [['1' => ''], ['1' => '0'], ['1' => '2'], ['1' => '3'], ['1' => '8']]);
```

Of course, there are other ways you could slice this... I'm just thinking the "array of arrays" approach might be more flexible since it would allow you to specify patterns that match two indicators at once if that need should arise.

Comment (demiankatz) [reply module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: I guess you could set things up so the array could include an array of values, meaning "any of these" and that would reduce it down to:

```
return $this->getMarcFieldWithInd('520', null, [['1' => ['', '0', '2', '3', '8']]]);
```

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: Since this doesn't involve MARC at all, you could move it up to DefaultRecord.

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: I'm not sure where the `getTopics` method comes from -- it's used in the RIS template but not defined anywhere. Seems like a local customization somewhere that crept into the core code.

I don't particularly like this name, since in faceting, we differentiate between topical subject headings and others like region or genre, but "getAllSubjectHeadings" should return all of the headings. Maybe it would be better to move this up to DefaultRecord, rename it something like "getAllSubjectHeadingsFlattened" and use that instead of getTopics in the RIS template to get rid of the non-standard method reference.

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (demiankatz) [inline themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: Maybe it would be better to add wrapper methods to the EDS driver to make the method names consistent instead of creating special alternate cases here.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @damien-git -- this looks like a significant improvement. I have a few suggested adjustments/improvements; see below.

File: module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php
Comment (damien-git) [reply module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: OK, I will revert that. I think the reason we added `$c` on our side was for other uses of that function (see this [example](https://catalog.lib.msu.edu/Record/folio.in00000100791?sid=4170223#details)).

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (damien-git) [reply module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: Funny, there is already a "`getCallNumbers()`" there which I had not noticed (with a similar implementation). Could we add a "`array_unique()`" to it to avoid duplicates ?

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (damien-git) [reply module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: OK

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [reply module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: Yes, I can't see a reason why that would be a problem!

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (damien-git) [reply themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: OK

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (meganschanz) [reply module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: Good suggestion! I did that refactoring and added some unit tests to get SolrMarc.php back to 100% coverage.

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (demiankatz) [inline themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: Would it make sense to move this logic into an EDS-specific implementation of getAllSubjectHeadingsFlattened to keep things simple and consistent?

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (demiankatz) [inline themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: Why is it necessary to strip tags from the DOI? If getCleanDOI is unexpectedly returning HTML, that seems like a problem we might need to fix at the record driver level rather than here.

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: I think you're missing some brackets in this example:

```suggestion
     *                         ex: [['1' => ['1', '2'], '2' => ['']]]
```

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: You don't need to indent parameter comments to line up like this -- you can wrap it all the way to the left * in order to make a little more room and require fewer lines.

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: I think it probably makes sense to move all of this new code into MarcAdvancedTrait so it can be more easily reused.

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: You should reflect nullability in the comment to match the signature:

```suggestion
     * @param ?array $subfield Sub-fields to return or empty for all
```

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: To match the documentation, shouldn't this be:
```suggestion
        return $this->getMarcFieldWithInd('520', null, [[1 => ['', '0', '2', '3', '8']]]);
```

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: Similarly, I think this should be:

```suggestion
        return $this->getMarcFieldWithInd('544', null, [[1 => ['', '0']]]);
```

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: I don't think this is quite right -- I think there's a potential for it to cause data to get duplicated (if both indicators match) or to be added based on a partial match (if one indicator matches but the other doesn't).

I think to make this all a bit easier to follow, maybe the indicator checking should be refactored to support methods, like this:

```
protected function checkIndicatorFilter($marc_data, $indFilter): bool
{
    foreach (range(1, 2) as $indNum) {
        if (isset($indData[$indNum])) {
            if (!in_array(trim(($marc_data['i' . $indNum] ?? '')), (array)$indData[$indNum])) {
                return false;
            }
        }
    }
    // If we got this far, no non-matching filters were encountered.
    return true;
}

protected function checkIndicatorFilters($marc_data, $indData): bool
{
    foreach ($indData as $indFilter) {
        if ($this->checkIndicatorFilters($marc_data, $indFilter)) {
            return true;
        }
    }
    // If we got this far, either $indData is empty (no filters defined -- return true)
    // or it is non-empty (no filters matched -- return false)
    return empty($indData);
}
```

Note that that's totally untested code, so I can't promise I got all the details right, but I hope you can see what I'm aiming for at least. The idea is that then the logic inside getMarcFieldWithInd() can be simplified to something like:

```
if ($this->checkIndicatorFilters($marc_data, $indData)) {
    $subfields = $marc_data['subfields'];
    foreach ($subfields as $subfield) {
        $field_vals[] = $subfield['data'];
    }
}
```

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @damien-git and @meganschanz! This is coming along nicely, though there are a few more details that may need tweaking...

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (damien-git) [reply themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: I was not sure if there was a semantic difference between the 2. With the RIS export, we can sometimes put data with slightly different meaning in the same key, because of the RIS format constraints. But in this case, it looks like `getItemsSubjects()` is only used for the RIS export, so we might as well replace it by `getAllSubjectHeadingsFlattened()` and return an array instead of a string (assuming nobody else is relying on it - that might be something to add to the release notes).

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (damien-git) [reply themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: I was not sure if that method was supposed to return HTML, although I did think it was not particularly "clean". In any case, it does return HTML (I forgot if it was with EDS or FOLIO), but I am also not sure if some calling code relies on it. It is used in HTML templates...

Comment (demiankatz) [reply themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: The method is called `getCleanDOI` because it's supposed to return a raw DOI -- if it's returning something else, that's a bug. I would imagine this is probably related to the EDS code.

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (demiankatz) [reply themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: Sounds good!

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (damien-git) [reply themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: OK. It is EDS. [Here](https://catalog.lib.msu.edu/EdsRecord/rzh,178847317#details) is an example ("DigiSpan: Development and Evaluation of a Computer-Based, Adaptive Test of Short-Term Memory and Working Memory."). The HTML is in `Items - Data` with `Name: "DOI"`:
`&lt;link linkTarget=&quot;URL&quot; linkWindow=&quot;10.1044/2024_JSLHR-23-00466&quot; linkTerm=&quot;https://www.doi.org/10.1044/2024_JSLHR-23-00466&quot;&gt;10.1044/2024_JSLHR-23-00466&lt;/link&gt;`
The same DOI is available in `RecordInfo - BibRecord - BibEntity - Identifiers` without HTML.

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (damien-git) [reply themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: done

File: themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml
Comment (damien-git) [reply themes/root/templates/RecordDriver/AbstractBase/export-ris.phtml]: fixed the EDS driver and updated the call

File: module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/EDSTest.php
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/EDSTest.php]: Do we really want HTML included in the flattened subject headings? Is it possible we want to keep `getItemsSubjects` as a mechanism for obtaining these HTML versions, and then make `getAllSubjectHeadingsFlattened` a wrapper around it that applies `strip_tags`?

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks for the progress, @damien-git. See below for one new question raised by the latest EDS refactoring.

File: module/VuFind/src/VuFind/RecordDriver/SolrMarc.php
Comment (meganschanz) [reply module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: I don't think duplicates could occur because of the `return array_unique($vals);` at the end of the function, but we could prevent them from being added by checking if it exists before adding to to vals:
```
$newVal = implode(' ', $field_vals);
if (!empty($field_vals) && !in_array($newVal, $vals)) {
    $vals[] = $newVal;
}
```

Our original hope with the function was to have it return if any of the indicator values matched, not only if they all matched for a field. For `getAbstractAndSummaryNotes` and `getLocationOfArchivalMaterialsNotes` it doesn't make a difference, but in our local code we have instances where we either indicator value to be returned. One of the unit tests check for that scenario:
```
// Test when different ind values are passed for each ind
$this->assertEquals(
    ['ger', 'spa'],
    $obj->getMarcFieldWithInd('041', null, ['1' => ['0'], '2' => ['7']])
);
```

I think re-wording the function comment to this would help:
```
ex: ['1' => ['1', '2'], '2' => ['']] would filter fields with ind1 = 1 or 2 or ind2 = blank
```

Comment (demiankatz) [reply module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: @meganschanz, my proposed implementation supports the ability to do AND or OR operations depending on your preferences... so, if you want to match BOTH indicators, you could pass in:

```
[['1' => ['1', '2'], '2' => ['']]]
```

...and if you want to match EITHER indicator, you could pass in:

```
[['1' => ['1', '2']], ['2' => ['']]]
```

Maybe this is an overengineered solution that we don't need, in which case you're certainly welcome to abandon it in favor of keeping it simple and adjusting the comments... but if you think the flexibility would be useful, it might be worth adapting as I proposed.

Comment (meganschanz) [reply module/VuFind/src/VuFind/RecordDriver/SolrMarc.php]: Ah ok, that makes sense. I like the idea of making it more flexible now to avoid inevitable refactoring later when we find a use for it. I'll work on that this afternoon and push another commit.

File: module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/EDSTest.php
Comment (damien-git) [reply module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/EDSTest.php]: I would prefer to avoid adding a function if we don't have a use case for it. How about simply applying `strip_tags` in `getAllSubjectHeadingsFlattened` ?

Comment (demiankatz) [reply module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/EDSTest.php]: Sure! The only reason I could think of keeping the historical function is in case somebody is relying on it for a local customization, but I suspect that's not especially likely. :-)

Comment (damien-git) [reply module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/EDSTest.php]: OK. I am removing the potential dot at the end while I'm at it.

File: module/VuFind/src/VuFind/RecordDriver/EDS.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/EDS.php]: Is it safe to explode on comma, or would we be better off doing some kind of regular expression extraction? I see from the example that there's a comma-separated list of `<a>` tags, but what happens if there's a comma inside the tag? It seems to me that this might be possible, though I can't be totally sure....

File: module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: Maybe we could simplify this to `@param array $indData Indicator filters as described in getMarcFieldWithInd()` so we don't have to include this long description and examples multiple times. Otherwise, this might get out of sync in future if changes are made.

File: module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: I expanded this a little bit in an effort to clarify:

```suggestion
     * @param array  $indData  Array of filter arrays, each in the format indicator number =>
     * array of allowed indicator values. If any one of the filter arrays fully matches the indicator
     * values in the field, data will be returned. If no filter arrays are defined, data will always
     * be returned regardless of indicators.
     * ex: [['1' => ['1', '2']], ['2' => ['']]] would filter fields ind1 = 1 or 2 or ind2 = blank
     * ex: [['1' => ['1'], '2' => ['7']]] would filter fields with ind1 = 1 and ind2 = 7
     * ex: [] would apply no filtering based on indicators
```

File: module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: Since we're adding these new methods, it would probably be a good idea to add them to the default RecordDataFormatter configuration so they can actually be used in places visible to the user as well as the data export -- I see no harm in having more visible notes in the description tab. That's probably out of scope for this PR, but maybe we can do that as a separate follow-up PR when this one is merged. What do you think?

File: module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/Feature/MarcAdvancedTraitTest.php
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/Feature/MarcAdvancedTraitTest.php]: I think the comments are reversed on this test and the following one -- it says OR here instead of AND, and vice versa.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks for the ongoing progress, @meganschanz and @damien-git! See below for a few more suggestions, mostly trivial and related to comments. (I also left a separate comment earlier today before this review regarding the extraction of EDS subjects).

File: module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php
Comment (meganschanz) [reply module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: I think that idea, but we probably should make it a separate PR since it does raise the question on if we would want them separated on the display. For example, in [our customization](https://github.com/MSU-Libraries/catalog/blob/main/vufind/module/Catalog/src/Catalog/RecordDriver/SolrMarc.php#L272-L326) we have the 520 field separated based on the indicator so they have different appropriate labels. For the RIS export it made sense to combine abstract and summary (as AB is defined as "Abstract or synopsis"), but we probably want them separate on the description tab.

If I make a PR after this one is done to add it to the display, do you have a preference on having the 520 split to separate fields like we have it or do you think just have it use what is in the new `getAbstractAndSummaryNotes` function? (If we split them, I think I'd update `getAbstractAndSummaryNotes` to call `getSummaryNotes` and `getAbstractNotes` to avoid duplication of logic).

Comment (demiankatz) [reply module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: I think it makes sense to do it in a more granular way and then make getAbstractAndSummaryNotes a wrapper as you propose. You could feel free to make the record driver changes here if you like!

File: module/VuFind/src/VuFind/RecordDriver/EDS.php
Comment (damien-git) [reply module/VuFind/src/VuFind/RecordDriver/EDS.php]: A valid concern. The problem is that the data we get from EDS is a mess. For instance, major subjects are separated with br tags, while minor subjects are separated with ";". VuFind is currently not parsing that correctly, so major and minor subjects are not displayed in the same way on the record page. Using the tags to try to separate entries is a good idea, maybe we can split on "`>; <`" or "`>, <`" instead. I wish we had better parsing of the data, structuring it and removing all HTML, but I don't want to get into that in the scope of that PR.

File: module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php
Comment (meganschanz) [reply module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: Unless you object I think I'd prefer making a separate PR since this one is focused on RIS export and not display. Then the new PR could include adding all the separate 520 fields to the description tab (and refactoring `getAbstractAndSummaryNotes` as described earlier). I can work on creating that one after this one is merged.

Comment (demiankatz) [reply module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: If that's how you prefer to do things, I don't object... but one other thought in case you like it: what if we drop getAbstractAndSummaryNotes entirely, and just use getSummaryNotes and getAbstractNotes in the RIS template? Then if another record driver supports only one or the other, the whole thing will still work... and we end up needing/using fewer methods.

File: module/VuFind/src/VuFind/RecordDriver/EDS.php
Comment (demiankatz) [reply module/VuFind/src/VuFind/RecordDriver/EDS.php]: What if you did something like `preg_replace('/>\s*[;,]\s*/', '>|', $str)` and then `explode('|', strip_tags($str))` on the result? That might cover more cases without breaking the integrity of the HTML in a way that would mess up the tag stripping...

File: module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php
Comment (meganschanz) [reply module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: Ok I added them to the display as part of this PR. I updated `getSummary` to `return $this->getMarcFieldWithInd('520', ['a'], [[1 => ['', '0', '2', '8']]]);` instead of `return $this->getFieldArray('520');`  to prevent the other separate 520 fields from being duplicated in there as well as their respective fields. The only reason I'm not having `getSummary` call `getSummaryNotes` was because I didn't think we'd want to affect the existing display that only displays the `a` subfield when calling `getSummary`.

Comment (demiankatz) [reply module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: That makes sense. A couple questions/comments, though:

1.) Is it possible we at least want to limit some of these to the alphabetic rather than numeric subfields? I'm not sure that we want to include things like URIs in raw user output or RIS export.

2.) I was suggesting adjusting the record driver methods here but leaving the actual display changes for a separate PR -- mainly because if we add new note fields to the display we also need to add the labels to the language files, and right now, we're in a language file freeze because release 10.1 is very close. If all of this work can wait to be merged until release 11.0, that's no problem... but if you wanted to try to get some of this into 10.1, we should split off any parts that require new language strings. Sorry for being a bit unclear about my intention before! :-)

Comment (meganschanz) [reply module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php]: Sounds good. I took out the changes to `RecordDataFormatterFactory.php` and the 520 fields not used in the RIS export -- we can add them back in later.

Agreed about the alphabetic subfields -- I made that update and updated the test fixture to make sure it gets excluded.

File: module/VuFind/src/VuFind/RecordDriver/EDS.php
Comment (damien-git) [reply module/VuFind/src/VuFind/RecordDriver/EDS.php]: I changed the implementation using your idea.

File: module/VuFind/src/VuFind/RecordDriver/EDS.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordDriver/EDS.php]: I think this comment is slightly out of date:

```suggestion
     * Get the abstract notes.
```

**Review by demiankatz — CHANGES_REQUESTED** [review]
One tiny comment adjustment, and then I think this is probably ready to merge.

Once this is merged, I suggest two separate follow-up PRs:

1.) Since this PR narrows the scope of getSummary (removing the Abstracts), I think we should add a separate 'Abstract' line to the RecordDataFormatter configuration as soon as possible, so users who upgrade don't lose visibility to any important information. Fortunately, we already have translations for an 'Abstract' label, so this can be done in a PR targeting the current dev branch without requiring new translation strings.

2.) The other notes you want to add should be introduced in a PR targeted against dev-11.0, since these will require new translations.

Please let me know if you have any questions or concerns. I'm running the full test suite on this branch right now and will report back when the run completes. I expect it should pass, and in that case, I think this is ready to merge once the trivial comment change below is addressed.

**Comment by demiankatz** [comment]
Tests are passing!

**Comment by meganschanz** [comment]
Should be all set now after the comment fix (good catch). I agree that 2 PRs makes sense for the follow up work. I'll start working on those today.

**Review by demiankatz — APPROVED** [review]
Thanks again, @damien-git and @meganschanz! I'm merging this now and will also ensure that the dev branch is merged into dev-11.0 so everything is up to date and ready for the follow-up work.

**Pull request merged**
